### ISSUE TITLE
Tasks status updates aren't reflected on graph

### ISSUE THREAD

**Issue by manjitapandey** [hotosm/fmtm#1420]
**Describe the bug**
Few tasks are locked on project page and when i view the number of tasks mapped and validated on infographics view, I see message saying "No tasks validated or mapped yet!"

**To Reproduce**

1. Go to 'any projects'
2. Map and validate few tasks
3. Go to infographs page and see mapped vs validated task graph

**Expected behavior**
The updates on tasks status should be reflected on Validated vs Mapped Task graph on infographics page. 

**Screenshots**
![image](https://github.com/hotosm/fmtm/assets/97273021/d085516c-e5aa-477e-9b1a-6479a7c115d2)

![image](https://github.com/hotosm/fmtm/assets/97273021/6b6e1f7f-a96f-4594-bea6-f1196f8d621a)

**Additional context**

**Comment by Sujanadh** [comment]
I think it is working well, you can see the project test workflow on the dev server to check it.

**Comment by Sujanadh** [comment]
May be that task status was updated before 30 days ago, perhaps? We return updates for the last 30 days, which is why that graph was empty.

**Comment by Sujanadh** [comment]
we can change the alt text to "No Tasks Validated or Mapped in the Last 30 Days", What do you think?

**Comment by manjitapandey** [comment]
> May be that task status was updated before 30 days ago, perhaps? We return updates for the last 30 days, which is why that graph was empty.

I don't think so. There was submissions on 17th of march which is less than 10 days. 

![image](https://github.com/hotosm/fmtm/assets/97273021/cafca5c0-996a-4e35-a085-058dd1b8dbab)

I couldn't reproduce the issue on dev server but it prevails on staging and prod server.

**Comment by spwoodcock** [comment]
In that case, I will merge the current dev to staging & if we stabilise the staging server, then we could make a patch release at some point next week 👍

v2024.2.1

**Comment by Sujanadh** [comment]
> > May be that task status was updated before 30 days ago, perhaps? We return updates for the last 30 days, which is why that graph was empty.
> 
> I don't think so. There was submissions on 17th of march which is less than 10 days.
> 
> ![image](https://private-user-images.githubusercontent.com/97273021/319454217-cafca5c0-996a-4e35-a085-058dd1b8dbab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIyOTY1MjYsIm5iZiI6MTcxMjI5NjIyNiwicGF0aCI6Ii85NzI3MzAyMS8zMTk0NTQyMTctY2FmY2E1YzAtOTk2YS00ZTM1LWEwODUtMDU4ZGQxYjhkYmFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA1VDA1NTAyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4NDYxOThkM2I1NDVjMGRlZGE3MGIyNjc3YmJhNmU2YjA0YzFmYWRmNGY4ZDY3ZjlhY2M2NDYwMzcxZDQ0NWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.M3Qr22XAd3Ws_34DGOlmaZdxq_H9GlnYjIVFOSMd-P8)
> 
> I couldn't reproduce the issue on dev server but it prevails on staging and prod server.

having submission wont update the task status. so graph is unrelated to submission but is related with task status history.

**Comment by Sujanadh** [comment]
what do you think about this? @spwoodcock  should we stick to this where contributors are who updates the task status or think of different approach to consider actual submitters?

**Comment by spwoodcock** [comment]
Actionable tasks:
- Update the alt text for the graph to show "No Tasks Validated or Mapped in the Last 30 Days"
- The submission count value should be determined by any task status that is >0
  - (0 is ready, anything higher means the task has been mapped, validated, or marked invalid).
  - Using a `>` should work as these are number values. If not, worse case we can do `!=0`.

**Comment by NSUWAL123** [comment]
@spwoodcock, maybe we should determine the submission count if the status is `>1` since the status `1` is for `locked for mapping` & it doesn't determine the submission of an entity. What are your thoughts?

**Comment by NSUWAL123** [comment]
After preparing data for the `validated vs mapped` chart from `/statuses` API in frontend, I realized that showing the `validated vs mapped` chart without an API call is not possible for now. 

Let's suppose I had mapped an entity, the status for that entity would be `2` after being mapped. Then the entity submission gets validated and the status of that entity is updated to `4` (overridden to 4). Now we don't have the history to show that the entity was mapped. 

@spwoodcock , @Sujanadh

**Comment by spwoodcock** [comment]
You make a good point, but in reality do we need the history if every validated task must have been mapped previously?

We know for sure that any task with status 4, must have been 2 beforehand

**Comment by NSUWAL123** [comment]
Umm. But let's say if the entity was mapped on `21 june` and validated on `26 june`, then the `updated_date` of the entity on the `/status `API would also be overridden by the `validated date` and the mapped date would be removed. So this would bring problem on the graph as there would be `mapped` line spike of June 21(before the entity gets validated) but after june 26(after the entity gets validated), the `mapped` line spike of June 21 would be flat raising visualization errors.

**Comment by spwoodcock** [comment]
Ah of course!
Thanks for clarifying!

Sorry if I'm asking dumb questions, as I'm replying from my phone in an airport...

The graph title talks about `tasks` rather than `features`, so surely this graph is for displaying the task history over time?

If this is the case, it's not relates to the entity statuses, but instead the task history endpoints, which include the date when each task had a changed status

**Comment by NSUWAL123** [comment]
Umm. I think the title should be updated to `Validated vs Mapped Submissions`

Whats your thought @manjitapandey ?

**Comment by NSUWAL123** [comment]
@spwoodcock, Sorry, I misunderstood that chart. The chart was related to tasks, not entities.

**Comment by NSUWAL123** [comment]
> @spwoodcock, maybe we should determine the submission count if the status is `>1` since the status `1` is for `locked for mapping` & it doesn't determine the submission of an entity. What are your thoughts?

@spwoodcock, I think its greater than 1 right?

**Comment by spwoodcock** [comment]
Yep you are right! 0 is ready, 1 is locked for mapping. Everything after has been mapped in some way

**Comment by manjitapandey** [comment]
@spwoodcock , @NSUWAL123 , Is conversation going on here is related to original issues mentioned?? If the conversation is about optimisation of API calls, we can shift our conversation to https://github.com/hotosm/fmtm/issues/1521. 
I feel the issue is already solved. 
Shall it be okay to close this issue for now??

**Comment by spwoodcock** [comment]
I think you are right - this can probably be closed!
